### ISSUE TITLE
Heizungsprofil HMIP in Programmen

### ISSUE THREAD

**Issue by eli4711** [jens-maus/RaspberryMatic#2444]
### Describe the issue you are experiencing

Das aktive Profil in Heizungsgruppen bei HMIP ist in Programmen auf 3 Profile beschränkt. In den HMIP Gruppen lassen sich aber 6 Profile (Wochenprogramme) einstellen.

![HMIP_Heating](https://github.com/jens-maus/RaspberryMatic/assets/40276118/275bc2ee-326d-44c9-b108-05d92867753f)

### Describe the behavior you expected

Alle 6 Profile lassen sich in Programmen auswählen.

### Steps to reproduce the issue

1. In einem Programm, eine Heizungsgruppe HMIP auf Profil 4 setzen lassen.
2. Es kommt direkt ein Popup mit dem Hinweis, Eingabe auf 1-3 beschränkt.
3. Oder eine Heizungsgruppe auf den Aktiven Status hin abfragen. (Wenn dann)

### What is the version this bug report is based on?

raspmatic_intelnuc (3.71.12.20230918)

### Which base platform are you running?

intelnuc (intel NUC)

### Which HomeMatic/homematicIP radio module are you using?

RPI-RF-MOD

### Anything in the logs that might be useful for us?

```txt
-
```

**Comment by jens-maus** [comment]
Um welches konkrete Gerät in dieser Heizungsgruppe geht es denn? Welche stecken da drin. Denn nicht ohne Grund kommt ja in der Konfiguration von HmIP-Heizungsgruppen bei der (?) neben der Auswahl des Wochenprogrammes zu dem Hinweis das nicht jedes Gerät die Wochenprofile 4-6 unterstützen. Siehe:

<img width="629" alt="Bildschirmfoto 2023-09-23 um 17 44 54" src="https://github.com/jens-maus/RaspberryMatic/assets/5887769/ee45deb2-6fc8-4f0a-9ba1-1178600fe8a1">

Es kommt also auf die konkreten Geräte an die in diesen Heizungsgruppen drinstecken ob die wirklich über bis zu 6 Wochenprofiles verfügen oder nicht. Daher bitte mal auflisten Welche Geräte in der gezeigten Gruppe drinstecken und dann in den jeweiligen Geräteeinstellungen dieser Geräte (Wandthermostat, Heizkörperventil, etc.) prüfen ob die auch wirklich bis zu 6 Wochenprofile zulassen.

**Comment by Baxxy13** [comment]
Es ist egal welches HmIP-Gerät (WTH / eTRV) in der Gruppe steckt.
`"MAX"` ist für `ACTIVE_PROFILE` immer `3`.

Das hat eQ-3 nicht richtig durchdacht.

Idealerweise wäre `"MAX"` per default `6`, und wenn man ein eTRV in die Gruppe packt müsste der auf `3` geändert werden.

**Comment by jens-maus** [comment]
Die eigentliche Frage wäre doch: Welches Gerät kann überhaupt 6 Wochenprofile beherbergen?

**Comment by Baxxy13** [comment]
Jedes WTH.
Also für 2-Punkt Regelung z.B. 6 Heizprofile oder in Verbindung mit FALMOT 3x Heizen 3x Kühlen.
Nur die eTRV's haben nur 3 (und vllt. WTH-B)

**Comment by jens-maus** [comment]
D.h. was? Gibt es jetzt das Problem oder nicht? Wenn ich also eine HmIP-Gruppe mit nur einem WTH und FALMOT habe sollte es erlaubt sein in Programmen bis max. 6 zu gehen, aber wenn in der Gruppe ein TRV drinsteckt der nur max. 3 Profile kann dann sollte er > 3 nicht zulassen? Ergo ist hier also nur die WebUI nicht smart genug das zu unterscheiden? Dann wäre das ein upstream bug in der WebUI soweit ich das verstehe. Richtig?

**Comment by Baxxy13** [comment]
Ich habe kein Problem damit. 

> Wenn ich also eine HmIP-Gruppe mit nur einem WTH und FALMOT habe sollte es erlaubt sein in Programmen bis max. 6 zu gehen

Ja.

> aber wenn in der Gruppe ein TRV drinsteckt der nur max. 3 Profile kann dann sollte er > 3 nicht zulassen?

Genau.

> Ergo ist hier also nur die WebUI nicht smart genug das zu unterscheiden?

So sieht das für mich aus.

Aber vielleicht kommt dazu ja noch was erläuterndes von @eli4711 

Ein Bug in RM ist es jedenfalls nicht. 😉

**Comment by Baxxy13** [comment]
Das Limit steckt in den META-Daten zum Datenpunkt "ACTIVE_PROFILE".
Die "Werte" stecken in der Device-Description im HmIP-Server. 
Die Gruppengeräte werden bei jedem Start "neu erzeugt" so das ein Ändern der Meta-Daten den Reboot nicht überlebt.


> dann schauen wir was er dann macht wenn in der Gruppe ein TRV mit max. 3 Profilen drinsteckt

Das eTRV schluckt Werte >3 nicht und bleibt beim aktuellen Profil.
D.h. man würde eine Inkonsistenz bei den Wochenprofilen über die Geräte erzeugen.

**Comment by eli4711** [comment]
In dem Fall ist es kein Bug in der RM. Hab die HmIP-eTRV-2 und HmIP-WTH-2 in der Gruppe. Da muss wohl eQ3 mal ran. 😉
Aber trotzdem Danke für eure Mühe! Und mal Ehrlich, die einzige Chance für Lösungen mit Problemen mit Homematic, sind hier (und im Forum) von RM vertreten. 😉

**Comment by jens-maus** [comment]
Ah ok, verstehe, na dann ist es erst recht ein upstream bug den wir wohl leider nicht selber beheben können und nur eQ3 da an den HMIPServer Hand anlegen müsste. Also doch nicht nur ein reiner WebUI Bug. Schade.

**Comment by Baxxy13** [comment]
In deinem Fall ist das genau richtig so. 
Wie gesagt können eTRV's nur 3 Profile.
Läuft das WTH dann mit Profil 4 und das eTRV mit 3 arbeiten die möglicherweise gegeneinander.

**Comment by eli4711** [comment]
Ja, hab ich auch gerade erst gesehen. In der Virtuellen Gruppe werden 6 Profile angeboten (und auch Änderungen akzeptiert), im eTRV-2 sind aber nur 3 vorhanden. Schande über mich, das mir das nicht schon vor dem Issue aufgefallen ist. 😑 Fazit, man lernt nie aus! Trotzdem Danke an alle 😉 Ihr seid einfach die Besten! 😎
